plugins/by-name/persisted/default.nix

```diff
@@ -0,0 +1,103 @@
+{
+  lib,
+  ...
+}:
+let
+  inherit (lib.nixvim) defaultNullOpts;
+  inherit (lib) types;
+in
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "persisted";
+  originalName = "persisted.nvim";
+  package = "persisted-nvim";
+
+  maintainers = [ lib.maintainers.GaetanLepage ];
+
+  settingsOptions = {
+    autostart = defaultNullOpts.mkBool true ''
+      Whether to automatically start the plugin on load.
+    '';
+
+    should_save =
+      defaultNullOpts.mkRaw
+        ''
+          function()
+            return true
+          end
+        ''
+        ''
+          Function to determine if a session should be saved
+
+          ```lua
+          @type fun(): boolean
+          ```
+        '';
+
+    save_dir = defaultNullOpts.mkStr (lib.nixvim.literalLua "vim.fn.expand(vim.fn.stdpath('data') .. '/sessions/')") ''
+      Directory where session files are saved.
+    '';
+
+    follow_cwd = defaultNullOpts.mkBool true ''
+      Whether to change the session file to match any change in the cwd.
+    '';
+
+    use_git_branch = defaultNullOpts.mkBool false ''
+      Whether to include the git branch in the session file name.
+    '';
+
+    autoload = defaultNullOpts.mkBool false ''
+      Whether to automatically load the session for the cwd on Neovim startup.
+    '';
+
+    on_autoload_no_session = defaultNullOpts.mkRaw "function() end" ''
+
+      Function to run when `autoload = true` but there is no session to load
+
+      ```lua
+      @type fun(): any
+      ```
+    '';
+
+    allowed_dirs = defaultNullOpts.mkListOf types.str [ ] ''
+      List of dirs that the plugin will start and autoload from.
+    '';
+
+    ignored_dirs = defaultNullOpts.mkListOf types.str [ ] ''
+      List of dirs that are ignored for starting and autoloading.
+    '';
+
+    telescope = {
```

Comment:
If they have a telescope integration, it's nice to create an option to add it to list of extensions to load. See https://github.com/nix-community/nixvim/blob/e2f81c8e8e8baa28b100e0e43b721f16de6299d8/plugins/by-name/project-nvim/default.nix#L118-L127

Reply:
Looks like it would be `persisted` 
> require("telescope").load_extension("persisted")

Reply:
Good idea ! Done.